spec/ntheory_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+
+describe SymEngine do
+  before :each do
+  end
+
+  describe 'NTheory' do
+
+    describe '#gcd' do
+      context 'GCD of 2 and 4' do
+        it 'returns 2' do
+          f = SymEngine::gcd(2, 4)
+          expect(f).to eql(2)
+        end
+      end
+    end
+
+    describe '#lcm' do
+      context 'LCM of 2 and 4' do
+        it 'returns 4' do
+          f = SymEngine::lcm(2, 4)
+          expect(f).to eql(4)
+        end
+      end
+    end
+
+    describe '#nextprime' do
+      context 'NextPrime after 4' do
+        it 'returns 5' do
+          f = SymEngine::nextprime(4)
+          expect(f).to eql(5)
+        end
+      end
+    end
+
+    describe '#mod' do
+      context '5 mod 4' do
```

Comment:
IMHO symengine's behavior should be consistent across all languages. But again there shouldn't be any surprises for ruby users. If both are same then it's good.